lib/Myriad/RPC.pm

```diff
@@ -0,0 +1,98 @@
+package Myriad::RPC;
+
+use strict;
+use warnings;
+
+# VERSION
+
+use Sys::Hostname;
+use Future::AsyncAwait;
+use Object::Pad;
+use Log::Any qw($log);
+
+use Myriad::RPC::Message;
+
+class Myriad::RPC extends Myriad::Notifier;
+
+use experimental qw(signatures);
+
+use utf8;
+
+use Syntax::Keyword::Try;
+
+=encoding utf8;
+
+=head1 SYNOPSIS
+
+=head1 DESCRIPTION
+
+Myriad RPC implementation to serve the requests of the service clients.
+
+=cut
+
+has $redis;
+has $service;
+
+has $group_name;
+
+has $ryu;
+has $rpc_map;
+has $whoami;
+
+method ryu {$ryu}
+method rpc_map :lvalue {$rpc_map}
+
+method configure(%args) {
+    $redis = delete $args{redis} // die 'Redis Transport is required';
+    $service = delete $args{service} // die 'Service name is required';
+
+    $whoami = hostname;
+    $group_name = 'processors';
+}
+
+method _add_to_loop($loop) {
+    $self->add_child(
+        $ryu = Ryu::Async->new
+    );
+
+    $self->listen->retain();
+}
+
+async method listen {
```

Comment:
If the method isn't otherwise taking any parameters, it is good style to put an empty `()` on it, so the checker still checks that nothing was passed, otherwise it'll silently ignore any additional arguments it did receive.

```perl
async method listen() {
    ...
```